src/blog/2023/09/making-sense-of-your-industrial-data.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Making sense of your industrial data
+subtitle: A look at legacy industrial protocol data and making sense of it in an IIoT world.
+description: A look at legacy industrial protocol data and making sense of it in an IIoT world.
+date: 2023-09-08
+authors: ["stephen-mclaughlin"]
+image: "/blog/2023/09/images/industrial-legacy-data-blog-image.png"
+tags:
+    - posts
+    - node-red
+    - community
+---
+
+In today's rapidly evolving industrial landscape, the ability to collect, analyse, and act upon data is more critical than ever. The Industrial Internet of Things (IIoT) has ushered in a new era of connectivity and automation, but what about the wealth of data locked away in legacy industrial protocols like ModBus? How can we unlock and make sense of this valuable information in the IIoT era? That's where tools like Node-RED and node-red-contrib-buffer-parser come into play.
+
+<!--more-->
+
+### Legacy Industrial Data: ModBus
+
+ModBus is one of the most widely used industrial protocols. Originally developed in the late 1970s, it has been a popular choice for industrial communication ever since. However, its data format can be challenging to work with in the context of modern IIoT applications. ModBus typically represents data in 16-bit unsigned registers, making it necessary to convert this data into more usable formats like Signed integer, Float, Signed and Unsigned Long, String, or even individual bits.
+
+### A short primer on data types
+
+Before we dive into how to make sense of ModBus data, let's take a quick look at some of the common data types we have to deal with.
+
+#### 16-bit unsigned
+
+16-bit unsigned data is an integer that can only be positive. It can represent values from 0 to 65535. For example, the number 12345 is represented as `0x3039` in hexadecimal or `0011000000111001` in binary.
```

Comment:
Not sure why a hexadecimal is introduced here, I'd just explain you have 16 bits for the value, and it's 2^16 values, as 0 is included the maximum values is 65535 (or 2^16 - 1).